src/main/java/com/example/onlinebookstore/dto/BookDto.java

```diff
@@ -1,11 +1,22 @@
 package com.example.onlinebookstore.dto;
 
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.PositiveOrZero;
 import java.math.BigDecimal;
 
 public record BookDto(
```

Comment:
this DTO is used in responses (which means you return it to clients)

therefore you don't need validation here :)